open-company/open-company-web
# REFACTOR: Move team related actions

Card: https://trello.com/c/nOF8CduC

Moved all team related actions to the  team action creators and store. Clean the api namespace that had part of the invite and switch user type logic.
Actions:
```
:teams-get
:teams-loaded
:team-loaded
:team-roster-loaded
:channels-enumerate
:channels-enumerate/success
:invite-users
:invite-user/success
:invite-user/failed
:user-action
:user-action/complete   // removed
:email-domain-team-add
:email-domain-team-add/finish
:slack-team-add
```

### Tests

#### Testing `:teams-get`

Used to load all the user's related teams data. It's triggered initially when the auth settings data are loaded and then only if the data are not present in 3 other places: on share post show, on org settings show and on signup.

Test 1:
- go to /sign-up
- signup with email and password
- fill in your first and last name
- [x] before setting your company check that `:teams-get`, `:teams-loaded`, `:team-loaded` and `:team-roster-loaded` have been called
- set you company name and proceed
- [x] once you are redirected to the org dashboard check that the previous actions are called again

Test 2:
- with the above setup create a post
- share the previous post
- [x] is slack disabled? Good
- [x] do you see a share post link in URL? Good

Test 3:
- go to Team Settings from the user menu
- [x] add an email domain, worked? No error in the console? Good
- [x] remove the above email domain, worked? No error in the console? Good

Test 4:
- go to invite users
- invite 3 users: one admin, one contributor and one viewer
- [x] do you see them in the Team settings panel with the right roles? Good
- switch the role: admin->viewer
- [x] do you see at least 2 DELETE call (one to remove admin and one to remove author)? Good
- switch the role: viewer -> admin
- [x] do you see a PUT and a POST request? Good
- switch the role: author -> admin
- [x] do you see 2, a PUT and a POST request? Good
- switch the admin to author again
- switch from author to viewer
- [x] do you see one DELETE request? Good

Test 5:
- resend one or 2 of the invitations
- [x] did it work? No errors? Good
- cancel at least one invitation
- [x] do you see a delete request? GOOD
- now remember (or take note) of the least of users with the role they have
- refresh the page and go back to the team settings
- [x] still the same list of users with the same roles? Good

Test 6:
- add a Slack org
- add the bot
- go to invite people
- [x] invite a slack user, worked? Good

Test 7:
- share a post
- [x] do you see slack now? Good
- [x] can you pick a slack channel? Good
- [x] share, worked without errors? Good

Test 8:
- go to team settings panel
- [x] remove the slack org from your team
- [x] worked? Good
